### ISSUE TITLE
[Wasm][Interp] Assertion at sgen-mono.c:114, condition `m_class_is_valuetype (klass)' not met

### ISSUE THREAD

**Issue by jeromelaban** [dotnet/runtime#52735]
### Description

The runtime fails to execute some code with the following trace:

```
dotnet.js:1601 * Assertion at /__w/runtime/src/mono/mono/metadata/sgen-mono.c:114, condition `m_class_is_valuetype (klass)' not met
607778    @    dotnet.js:1601
_emscripten_asm_const_int    @    dotnet.js:6347
wasm_trace_logger    @    00d55d3e:0x1e593a
eglib_log_adapter    @    00d55d3e:0xb281b
monoeg_g_logstr    @    00d55d3e:0x107cc8
monoeg_g_logv_nofree    @    00d55d3e:0x107c6c
monoeg_assertion_message    @    00d55d3e:0x107d16
mono_assertion_message    @    00d55d3e:0x107d4d
mono_gc_wbarrier_value_copy_internal    @    00d55d3e:0xacea6
mono_value_copy_internal    @    00d55d3e:0x8d52d
interp_exec_method    @    00d55d3e:0x1328f
interp_runtime_invoke    @    00d55d3e:0xa809
mono_jit_runtime_invoke    @    00d55d3e:0xd8e93
do_runtime_invoke    @    00d55d3e:0x8751d
mono_runtime_try_invoke    @    00d55d3e:0x87bcd
mono_runtime_invoke    @    00d55d3e:0x89ec8
mono_wasm_invoke_method    @    00d55d3e:0x1e5a35
```

Does not fail with Interp+AOT.
Repro: https://github.com/jeromelaban/Wasm.Samples/tree/master/Bug52735

### Configuration

Last working: da580beb984f40516a408b8270e08d93daa0364e
Failure since efd3410954bd8e014a51e0c35e1bd9be8571d77c (https://github.com/dotnet/runtime/pull/52130)

### Regression?

Yes

### Other information

https://github.com/dotnet/runtime/pull/52130

Trying to add more information, here's the log for this:
![image](https://user-images.githubusercontent.com/5839577/118208820-6d99e280-b435-11eb-8ab8-475f74c1d6c7.png)

Right before the assert:

```
Adding value remset at 0x166e20c, count 1, descr 0 for class ؂
dotnet.js:1 ؂
dotnet.js:1 𶕀
dotnet.js:1  (0xa55fd00)
```

**Comment by BrzVlad** [comment]
@jeromelaban Could I get info on how to reproduce this ? I doesn't matter if it's a specific or bloated app.

**Comment by jeromelaban** [comment]
I can provide you that, I'll try the PR as well so we can do this faster :) Thanks for the speed of fixing !

**Comment by jeromelaban** [comment]
Here's the repro: https://github.com/jeromelaban/Wasm.Samples/tree/master/Bug52735

Also, the fix in https://github.com/dotnet/runtime/pull/52752 does not seem to address the issue.

**Comment by jeromelaban** [comment]
Update, the issue *is* fixed by #52752.
